dev/README_AIRFLOW3_DEV.md

```diff
@@ -0,0 +1,114 @@
+<!--
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+-->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of contents**
+
+- [Main branch is Airflow 3](#main-branch-is-airflow-3)
+- [Contributors](#contributors)
+  - [Developing for providers and Helm chart](#developing-for-providers-and-helm-chart)
+  - [Developing for Airflow 3 and 2.10.x / 2.11.x](#developing-for-airflow-3-and-210x--211x)
+  - [Developing for Airflow 3](#developing-for-airflow-3)
+  - [Developing for Airflow 2.10.x](#developing-for-airflow-210x)
+  - [Developing for Airflow 2.11](#developing-for-airflow-211)
+- [Committers / PMCs](#committers--pmcs)
+  - [Merging PRs for providers and Helm chart](#merging-prs-for-providers-and-helm-chart)
+  - [Merging PR for Airflow 3 and 2.10.x / 2.11.x](#merging-pr-for-airflow-3-and-210x--211x)
+  - [Merging PRs 2.10.x](#merging-prs-210x)
+  - [Merging PRs for Airflow 3](#merging-prs-for-airflow-3)
+  - [Merging PRs for Airflow 2.11](#merging-prs-for-airflow-211)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Main branch is Airflow 3
+
+The main branch is for development of Airflow 3.
+Airflow 2.10.x releases will be cut from `v-2-10-stable` branch.
+
+# Contributors
+
+The following section explains which branches you should target your PR to
+
+## Developing for providers and Helm chart
+
+PRs should target `main` branch.
+Make sure your code is only about Providers or Helm chart.
+Avoid mixing core changes into the same PR
+
+## Developing for Airflow 3 and 2.10.x / 2.11.x
+
+If the PR is relevant for both Airflow 3 and 2 it should target `main` branch.
+
+## Developing for Airflow 3
+
+PRs should target `main` branch.
+
+## Developing for Airflow 2.10.x
+
+PR should target `v-2-10-test` branch. When cutting a new release for 2.10 release manager
+will sync `v-2-10-test`  branch to `v-2-10-stable` and cut the release from the stable branch.
+PR should never target `v-2-10-stable` unless specifically instructed by release manager.
+
+## Developing for Airflow 2.11
+
+Version 2.11 is planned to be cut from `v-2-10-stable` branch.
```

Comment:
I believe this one needs clarification - and it's very related to the questions that @uranusjr posted in https://lists.apache.org/thread/56b7bdrzszkqtd6kgy360okw72568ld2

Especially the point:

> - It also seems unproductive to hold off merging the PR until the 2.10 line is finished?

I think we need to have a way to somehow make it manageable to backport/have 2.11 - targetting PRs only.

This is missing now - we might have some PRs that are specifically targetting Airflow 2.11 and we do not want to merge then in 2.10, but they also do not make sense for Airflow 3 / main.

My proposal would be to cut-off `v-2-11-test` branch now (from the tip of the `v-2-10-test` and periodically rebase it on top of `v2-10-test`. This way we could have **SOME** PRs implemented in `v2-11-test` branch only .

Reply:
To my understanding 2.11 is meant to be bridge release... Thus it will not have features other than what is needed for the bridge functionality. Thus cutting it from 2.10 branch make sense.

Reply:
As discussed - I am fine with this approach as long as we have an explicit separate thread explaining it / vote on devlist as so far this only have been discussed but never voted on and I think we should be very explicit about it.